addons/mail/static/src/models/messaging.js

```diff
@@ -170,13 +170,14 @@ registerModel({
             } else {
                 const { args, method, model, kwargs = {} } = params;
                 const { domain, fields, groupBy } = kwargs;
+                delete kwargs.fields;
```

Comment:
I think we should remove it manually from each case that needs it to be removed, I don't think the default call should have anything removed.

Reply:
> I think we should remove it manually from each case that needs it to be removed, I don't think the default call should have anything removed.

Done.
FYI-
 In this case, it will fail.
```await this.messaging.rpc({
        model: 'res.users',
        method: 'search',
        args: [[['id', '=', 1]]],
        kwargs: {
              context,
              fields,
        },
     }, { shadow: true }); 
```

I have not checked other cases.

Reply:
Thank you for checking, if you know `search` has the bug too you could fix it too please!

And verify all the other methods so that we sure all of them are fine.

Reply:
> Thank you for checking, if you know `search` has the bug too you could fix it too please!
> 
> And verify all the other methods so that we sure all of them are fine.


 the issue occurs in these  method- read, unlink, search, write. now, what to do? 

```
unlink: 
    await this.messaging.rpc({
        model: 'hr.employee',
        method: "unlink",
        args: [ids],
        kwargs: {
            fields,
        },
    });

search:
   await this.messaging.rpc({
        model: 'hr.employee',
        method: 'search',
        args: [[['id', '=', 1]]],
        kwargs: {
              fields,
        },
     }); 

write:
    await this.messaging.rpc({
        model: 'hr.employee',
        method: "write",
        args: [ids, [{'name': 'ppr'}],
        kwargs: {
            fields,
        },
    });  
```